src/components/JournalIndexPreview/index.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import JournalIndexPage from '../JournalIndexPage';
+
+class JournalIndexPreview extends React.Component {
+  componentDidMount() {
+    this.props.getPreview(this.props.match.params.previewId);
+    this.props.getJournals();
+  }
+  componentDidUpdate(prevProps) {
+    if (prevProps.match.params.previewId !== this.props.match.params.previewId) {
+      this.props.getPreview(this.props.match.params.previewId);
```

Comment:
Is this a use case that can happen (being on one index page and being redirected to another inside the app)?